iGCS/Models/Craft/GCSCraftArduCopter.m

```diff
@@ -30,11 +37,34 @@ @implementation GCSCraftArduCopter
         _guidedMode  = APMCopterGuided;
         _setModeBeforeGuidedItems = YES; // For 3.2+
         _icon = [UIImage imageNamed:@"quad-icon-128.png"];
+        _notificationCenter = [NSNotificationCenter defaultCenter];
+
+        // set heartbeat to nil when app goes into background so all
+        // events fire again on app launch and telemetry reconnect
+        __weak GCSCraftArduCopter *weakSelf = self;
+        [self.notificationCenter addObserverForName:UIApplicationDidEnterBackgroundNotification
+                                             object:nil
+                                              queue:self.craftQueue
+                                         usingBlock:^(NSNotification *note) {
+                                             [weakSelf clearHearbeat];
+                                         }];
+
+        // set heartbeat to nil when app loses telemetry for a period of time
+        [self.notificationCenter addObserverForName:GCSHeartbeatManagerHeartbeatWasLost
+                                             object:nil
+                                              queue:self.craftQueue
+                                         usingBlock:^(NSNotification *note) {
+                                             [weakSelf clearHearbeat];
+                                         }];
     }
 
     return self;
 }
 
+- (void)dealloc {
```

Comment:
Ok, will stop with the DRYs. We have a class hierarchy here, so let's express it as one (especially before we end up with additional craft models)